modin/experimental/engines/omnisci_on_native/test/utils.py

```diff
@@ -66,6 +71,40 @@ def omnisci_comparator(df1, df2):
     )
 
 
+def align_datetime_dtypes(*dfs):
+    """
+    Make all of the passed frames have DateTime dtype for the same columns.
+
+    Cast column type of the certain frame to the DateTime type if any frame in
+    the `dfs` sequence has DateTime type for this column.
+
+    Parameters
+    ----------
+    *dfs : iterable of DataFrames
+        DataFrames to align DateTime dtypes.
+
+    Notes
+    -----
+    Passed Modin frames may be casted to pandas in the result.
+    """
+    datetime_cols = {}
+    for df_dtypes in zip(*tuple(df.dtypes.items() for df in dfs)):
```

Comment:
Why do you want to make an explicit `tuple()` here instead of just using an iterator? `zip()` is an iterator itself, so I suggest to not materialize everything but do it in iterative way (probably just dropping `tuple` word here should work).

This also feels like it's trying to be clever when it's not needed. I suggest to change to something like
```python
for df in dfs:
    for col, dtype in df.dtypes.items():
```
same amount of `for` loops, same amount of lines, but reads much easier IMHO

Reply:
it seems that I've overcomplicated things a bit 😄 

applied your suggestion